Cpt jack04/ui and audio improvements

clinealex10 opened the pull request.

clinealex10 committed:
Worked to reduce scene dependancies for the scenes to work probably.

- Pause Menu Manager is now a ScriptableObject.
- Pause Menu is now loaded dynamically into any scene whose name doesn't contain "Menu_"
- Pause Menu includes how to play and settings. Both are functional. There is a little flash when enabling and disabling these canvases using the Esc key but to be honest, it's fine for now.
- The Game Manager now knows the first level by build index.
- There's probably some other stuff too but I forgot ¯\_(ツ)_/¯

clinealex10 committed:
Made some small audio changes

NightAngel47 committed:
Made Updated Pause Menu a Prefab

Old pause menu prefab wasn't up to date, so I removed the old one and made the updated pause menu a new prefab.

NightAngel47 committed:
Delete Plugins.meta

NightAngel47 reviewed (changes requested):
Reviewing your PR and when testing it, I noticed that when you open the settings menu on the pause screen after loading a level, the sliders are all set to max which resets all of the volume sliders from whatever was set. This however, only happens the first time during a session. 

I also noticed that the theme music plays during the ending poem. Idk if that was intentional.

clinealex10 committed:
Music playing at the end is fixed. Volume reset bug isn't.

I have an idea of what's causing the problem with the volume resetting in the pause menu, but I can't consistantly do it enough to test for it.
We'll just have to keep our eye on it.

clinealex10 committed:
Started a custom inspector for MoonlightBehaviour

I didn't feel like waiting til git was fixed so I made on the custom inspector.

clinealex10 committed:
Removed Boo.Lang from VolumeManager.cs
